src/vllm_tgis_adapter/__main__.py

```diff
@@ -328,41 +61,30 @@ async def run_http_server(
         usage_context=UsageContext.OPENAI_API_SERVER,
     )
 
-    event_loop: asyncio.AbstractEventLoop | None
-    try:
-        event_loop = asyncio.get_running_loop()
-    except RuntimeError:
-        event_loop = None
-
-    if event_loop is not None and event_loop.is_running():
-        # If the current is instanced by Ray Serve,
-        # there is already a running event loop
-        model_config = event_loop.run_until_complete(engine.get_model_config())
-    else:
-        # When using single vLLM without engine_use_ray
-        model_config = asyncio.run(engine.get_model_config())
+    loop = asyncio.new_event_loop()
+    tasks = loop.run_until_complete(start_servers(args))
 
-    if event_loop is None:
-        event_loop = asyncio.new_event_loop()
+    def signal_handler() -> None:
+        # prevents the uvicorn signal handler to exit early
+        for task in tasks:
+            task.cancel()
 
-    async def run() -> None:
+    async def override_signal_handler() -> None:
         loop = asyncio.get_running_loop()
-
-        http_server_task = loop.create_task(run_http_server(engine, args, model_config))
-        grpc_server_task = loop.create_task(
-            run_grpc_server(
-                engine, args, disable_log_stats=engine_args.disable_log_stats
-            )
-        )
-
-        def signal_handler() -> None:
-            # prevents the uvicorn signal handler to exit early
-            grpc_server_task.cancel()
-            http_server_task.cancel()
-
-        loop.add_signal_handler(signal.SIGINT, signal_handler)
-        loop.add_signal_handler(signal.SIGTERM, signal_handler)
-
-        await asyncio.gather(grpc_server_task, http_server_task)
-
-    event_loop.run_until_complete(run())
+        while not loop._signal_handlers:  # noqa: SLF001
+            logger.debug("No signal handlers are defined, sleeping")
+            await asyncio.sleep(1)
+
+        logger.debug("Signal handlers have been setup, proceeding with override")
```

Comment:
This trick is required because the signal handlers are set by `vllm.entrypoints.launcher.serve_http`, which are set up at a later stage: we have to wait for them to be set up in order to ovveride them and cleanly shut down the servers.

Reply:
I'm not sure about the way that async and non-async is mixed here, but it if works that's fine, we can always refactor it a bit later.